packages/eslint-plugin-pf-codemods/src/rules/v6/noDuplicateImportSpecifiers/no-duplicate-import-specifiers.ts

```diff
@@ -0,0 +1,78 @@
+import { Rule } from "eslint";
+import { ImportDeclaration, ImportSpecifier } from "estree-jsx";
+import { getFromPackage, removeSpecifierFromDeclaration } from "../../helpers";
+
+// Cleanup from other rules
+module.exports = {
+  meta: { fixable: "code" },
+  create: function (context: Rule.RuleContext) {
+    const { imports } = getFromPackage(context, "@patternfly/react-core");
+
+    const createKey = (specifier: ImportSpecifier) =>
+      `${specifier.imported.name}:${specifier.local.name}`;
+
+    const findDuplicates = (specifiers: ImportSpecifier[]) => {
+      const occurrences = new Map<string, number>();
+
+      // Build a map to count occurrences of each similar import
+      specifiers.forEach((specifier) => {
+        const key = createKey(specifier);
+        if (occurrences.has(key)) {
+          occurrences.set(key, occurrences.get(key)! + 1);
+        } else {
+          occurrences.set(key, 1);
+        }
+      });
+
+      const seen = new Set<string>();
+      const duplicates: ImportSpecifier[] = [];
+
+      specifiers.forEach((specifier) => {
+        const key = createKey(specifier);
+
+        if (occurrences.get(key) === 1) {
+          return;
+        }
+
+        if (seen.has(key)) {
+          duplicates.push(specifier);
+        } else {
+          seen.add(key);
+        }
+      });
+
+      return duplicates;
+    };
```

Comment:
This is very cool, but also feels a bit complex for what we're really doing here.

Could we do something like:
```
    const findDuplicates = (specifiers: ImportSpecifier[]) => {
      const localNames = specifiers.map((spec) => spec.local.name);

      return specifiers.filter(
        (specifier, index) => localNames.indexOf(specifier.local.name) !== index
      );
    };

Reply:
Oh, that looks much nicer. 

I was for some reason stuck on a mindset, that both `imported` and `local` names must match. Because if consumer would have something like: `import { Button as ABC, Text as ABC }` we would remove an import of a component we don't want to remove - but such code is invalid in the first place, so we can work with local names only.